ocaml/src/nsm_model.ml

```diff
@@ -32,6 +32,23 @@ module OsdInfo = struct
   type ip = string [@@deriving show, yojson]
   type port = int [@@deriving show, yojson]
 
+  type endpoint_uri = string [@@deriving show, yojson]
+
+  type host = string
+  type endpoint = Net_fd.transport * host * port
+
+  let parse_endpoint_uri s =
+    let uri = Uri.of_string s in
+    let transport = match Uri.scheme uri with
+      | None -> failwith "Transport should be 'tcp' or 'rdma'"
+      | Some "tcp" -> Net_fd.TCP
```

Comment:
ok.  from [RFC3986](https://tools.ietf.org/html/rfc3986:)
```
Although schemes are case-insensitive, the canonical form is lowercase and 
documents that specify schemes must do so with lowercase letters.  
An implementation should accept uppercase letters as equivalent to lowercase 
in scheme names (e.g., allow "HTTP" as well as "http") for the sake of
robustness but should only produce lowercase scheme names for consistency.
```
The library does the right thing ;)